core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import com.spatial4j.core.io.GeohashUtils;
+
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.geo.GeoPointFieldMapper;
+import org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.Type;
+import org.elasticsearch.index.search.geo.IndexedGeoBoundingBoxQuery;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MIN_LAT;
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MAX_LAT;
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MIN_LON;
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MAX_LON;
+
+@SuppressWarnings("deprecation")
```

Comment:
Not sure about surpressing deprecation, is this because testing all these deprecated setters? I think it would be good to see them even if it is annoying to remember deleting them at some point.